src/main/java/org/folio/circulationbff/service/impl/CirculationLoanServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package org.folio.circulationbff.service.impl;
+
+import static java.lang.Boolean.TRUE;
+import static java.util.stream.Collectors.toMap;
+import static org.folio.circulationbff.domain.mapping.CirculationLoanMapper.toStream;
+
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.ListUtils;
+import org.folio.circulationbff.client.feign.CirculationClient;
+import org.folio.circulationbff.domain.dto.BffSearchInstance;
+import org.folio.circulationbff.domain.dto.BffSearchItem;
+import org.folio.circulationbff.domain.dto.CirculationLoan;
+import org.folio.circulationbff.domain.dto.CirculationLoans;
+import org.folio.circulationbff.domain.dto.LoanItem;
+import org.folio.circulationbff.domain.mapping.CirculationLoanMapper;
+import org.folio.circulationbff.service.CirculationLoanService;
+import org.folio.circulationbff.service.SearchService;
+import org.folio.circulationbff.service.SettingsService;
+import org.folio.circulationbff.service.TenantService;
+import org.folio.circulationbff.support.CqlQuery;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class CirculationLoanServiceImpl implements CirculationLoanService {
+
+  private final SearchService searchService;
+  private final TenantService tenantService;
+  private final SettingsService settingsService;
+  private final CirculationClient circulationClient;
+  private final CirculationLoanMapper circulationLoanMapper;
+
+  @Override
+  public CirculationLoans findCirculationLoans(String query, Integer limit,
+    Integer offset, String totalRecords) {
+    var circulationLoans = circulationClient.findLoansByQuery(query, limit, offset, totalRecords);
+
+    var dcbItemIds = circulationLoans.getLoans().stream()
+      .filter(CirculationLoanServiceImpl::shouldEnrichCirculationLoan)
+      .map(CirculationLoan::getItemId)
+      .toList();
+
+    if (CollectionUtils.isEmpty(dcbItemIds) || !isEnrichOperationAllowed()) {
+      return circulationLoans;
+    }
+
+    var bffInstances = getBffInstancesByItemId(dcbItemIds);
+
+    for (var loan : circulationLoans.getLoans()) {
+      if (shouldEnrichCirculationLoan(loan)) {
```

Comment:
I think it would be beneficial for debugging purposes to log.info this if (we're trying to log most of the `if`s, actually). And maybe even outside of the if as well, to see which loan records weren't enriched

Reply:
Added log message at DEBUG level, because for info it will be too informative, also, all enriched loan IDs will be printed, so it will indicate what has been enriched, and what is not